text/0033-private-testnet-guide.md

```diff
@@ -0,0 +1,424 @@
+```
+
+This command configures the fullnode to:
+- Expose its HTTP api on port 8082
+- Connect to `tcp://127.0.0.1:40403` (fullnode-explorer) to sync with the P2P network
+- Use `~/hathor-private-tutorial/fullnode-wallet-2` as its data directory in the host.
+- Use the configuration file we create for the network in `hathor.conf.privnet`
+- Enable `wallet index`, which is needed to perform some kinds of operations in the api, like getting transactions information.
+
+### Fullnode Miner
+This fullnode will be used by our miner, and we will call it `fullnode-miner` for that reason.
+
+```sh
+mkdir -p ~/hathor-private-tutorial/fullnode-miner
+
+docker run -ti --network="host" -v ~/hathor-private-tutorial/fullnode-miner:/data:consistent -v ~/hathor-private-tutorial/conf:/privnet/conf --env PYTHON_PATH=/privnet/conf --env HATHOR_CONFIG_FILE=privnet.conf.privnet hathornetwork/hathor-core run_node --cache --status 8083 --stratum 8093 --data /data --x-fast-init-beta --bootstrap tcp://127.0.0.1:40403
+```
+
+This command configures the fullnode to:
+- Expose its HTTP api on port 8083
+- Expose its miing api on port 8093
+- Connect to `tcp://127.0.0.1:40403` (fullnode-explorer) to sync with the P2P network
+- Use `~/hathor-private-tutorial/fullnode-miner` as its data directory in the host.
+- Use the configuration file we create for the network in `hathor.conf.privnet`
+
+# Running the miner
+Now that our fullnodes are in place, the next step is to have a miner to mine blocks in our new private network.
+
+The easiest way to do this in a local testing environment is to run a CPU Miner.
+We have a CPU Miner as a Docker image in https://hub.docker.com/r/hathornetwork/cpuminer
+
+You should run it with the following command, replacing `<address>` by the same address we generated in the beginning of the guide:
+```sh
+docker run -ti --network="host" hathornetwork/cpuminer -t 1 -a sha256d -o stratum+tcp://127.0.0.1:8093 --coinbase-addr <address>
+```
+
+This command is telling the miner to connect to our `fullnode-miner` in the port 8093.
+
+When you start mining, you'll see that all fullnodes start logging the new blocks being received. The logs of the `fullnode-miner` will be different, because the miner will be connected directly to it sending mined blocks. The other fullnodes will only receive the new blocks through the sync algorithm.
+
+Also note that all fullnodes will receive the blocks, even though only the `fullnode-explorer` is directly connected to the `fullnode-miner`.
+
+# Running a TxMiningService
+The role of this service is to act as a miner for transactions generated by wallets. In Hathor, transactions are also mined just like blocks, just with a lower difficulty.
+
+For this one, the repo has to be cloned from https://github.com/HathorNetwork/tx-mining-service
```

Comment:
We now have Docker images for tx-mining-service in Docker Hub.

This should be updated to use them, instead of cloning the repo and building them locally.